### DISCUSSION TITLE
Bleak issue of time to display data

### DISCUSSION THREAD

**Discussion by cha99cg** [hbldh/bleak#1599] [Q&A]
* bleak version:
* Python version:
* Operating System:
* BlueZ version (`bluetoothctl -v`) in case of Linux:

### Description
i'm using bleak to just recover distance every 50ms in fact the recovery is in 200ms or more.
the distance is between two shields NXP.

### What I Did

i did a script to just display the real distance like this
```python
import asyncio
import timeit
from bleak import BleakScanner
from bleak.backends.scanner import AdvertisementData
import csv
output = 'Recup_Data.csv'
from bleak.backends.scanner import AdvertisementData
with open(output, 'w', newline='') as f:
    writer = csv.writer(f)
    writer.writerow(["Counter", "Distance","Adresse MAC"])
last_counter = None
timeit.timeit(lambda: "-".join(map(str, range(100))), number=10000)
async def get_advertisement_data_between_macs():
    def detection_callback(device, advertisement_data: AdvertisementData):
        global last_counter
        if (advertisement_data.local_name == "W"):
            print(advertisement_data)
            print(advertisement_data.manufacturer_data[0xFFFF])
            counter = advertisement_data.manufacturer_data[0xFFFF][0] + (advertisement_data.manufacturer_data[0xFFFF][1] << 8)
            print("Compteur :",counter) 
            if counter != last_counter :
                last_counter = counter
                Mac_adrr = advertisement_data.manufacturer_data[0xFFFF][2:9] 
                mac_address_str = ':'.join(f'{b:02x}' for b in Mac_adrr)
                print("MAC Address :",mac_address_str )
                Distance = advertisement_data.manufacturer_data[0xFFFF][9] + (advertisement_data.manufacturer_data[0xFFFF][10] )
                print("Distance : " , Distance)           
                with open(output, 'a', newline='') as f:
                    writer = csv.writer(f)
                    writer.writerow([counter, Distance, mac_address_str])
    async with BleakScanner(detection_callback, scanning_mode ='passive') as scanner:
        while True:
            await asyncio.sleep(0.05)  
asyncio.run(get_advertisement_data_between_macs())
```

### Logs
every 200ms there is a new distance but i want iin every 50ms

**Comment by dlech** [comment]
Bleak can't tell a peripheral how often to send advertising data. 

Opening and closing the file in the detection callback is probably hurting performance.


> ```python
>         while True:
>             await asyncio.sleep(0.05)  
> ```

Tip, a more efficient way to wait forever is:

```python
    event = asyncio.Event()
    await event.wait()
```

**Reply by cha99cg** [reply]
thank you, but i know bleak can't tell that's why i did a software in MCUxpresso so that i leant on Bleak to just watch my distance in fact i totally but 50ms in the software that i did.

thank you in advance.

**Reply by cha99cg** [reply]
Even i did like this"event = asyncio.Event()
    await event.wait()"
it doesn't work i still do have two or three packets lost in every two consecutive distance.

thank you.

**Reply by dlech** [reply]
Did you remove `with open(output, 'a', newline='') as f:` and all of the `print`s from the detection callback? These functions all take a long time to run.
